TemplePlus/python/python_object.cpp

```diff
@@ -2443,6 +2444,11 @@ static PyObject* PyObjHandle_Unwield(PyObject* obj, PyObject* args) {
 		es = EquipSlot::Invalid;
 	auto equipSlot = (EquipSlot)es;
 
+	if (item != objHndl::null && inventory.GetParent(item) == self->handle ) {
```

Comment:
From what I see, the logic you propose is this:
If the optional arg used and is an item handle of **_any item that is in the wielder's inventory_**, then the unwield action should drop the item worn at slot **es**.

I don't understand the logic behind this, and it seems very "ad-hoc". It would be more intuitive for it to accept a flag (or integer) arg.